pubsub.go

```diff
@@ -848,7 +852,11 @@ type topicReq struct {
 // GetTopics returns the topics this node is subscribed to.
 func (p *PubSub) GetTopics() []string {
 	out := make(chan []string, 1)
-	p.getTopics <- &topicReq{resp: out}
+	select {
+	case p.getTopics <- &topicReq{resp: out}:
+	case <-p.ctx.Done():
+		return []string{}
```

Comment:
we can just return `nil` here.

Reply:
Figured that it would be less breaking to return empty then nil in case some code doesn't check for `nil`. If you think `nil` is better though just give the say so and I'll fix it.

Reply:
It's more idiomatic to return `nil`.